mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -173,6 +173,10 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	// 4. Header Transport - outer wrapper to inject additional headers we want to apply
 	// before making requests
 	headerTransport := newTransportWithHeaders(retryTransport)
+	// opt in extension for adding to the User-Agent header
+	if ext := os.Getenv("GOOGLE_TPG_USERAGENT_EXTENSION"); ext != "" {
+		headerTransport.Set("User-Agent", ext)
```

Comment:
Oh, sorry, I thought `providerConfigure` from provider.go was in config.go. Let's modify Config.userAgent there instead of adding new functionality to the transport.

Reply:
done